test/cell_test.rb

```diff
@@ -0,0 +1,81 @@
+require 'minitest/autorun'
+require 'minitest/pride'
+require './lib/cell'
+
+class CellTest < Minitest::Test
+
+  def setup
+    @cell = Cell.new("B4")
+  end
+
+  def test_it_exists
+    assert_instance_of Cell, @cell
+  end
+
+  def test_it_has_a_coordinate
+    assert_equal "B4", @cell.coordinate
+  end
+
+  def test_cell_is_empty
+    assert_equal nil, @cell.ship
+    assert_equal true, @cell.empty?
+  end
+
+  def test_a_ship_can_be_placed_on_cell
+    cruiser = Ship.new("Cruiser", 3)
+    @cell.place_ship(cruiser)
+
+    assert_equal cruiser, @cell.ship
+    assert_equal false, @cell.empty?
+  end
+
+  def test_cell_can_be_fired_upon
+    cruiser = Ship.new("Cruiser", 3)
+    @cell.place_ship(cruiser)
+
+    assert_equal false, @cell.fired_upon?
+
+    @cell.fire_upon
+
+    assert_equal 2, @cell.ship.health
+    assert_equal true, @cell.fired_opon?
+  end
+
+  def test_render_is_dynamic_for_miss
+
+    assert_equal ".", @cell.render
+
+    @cell.fire_upon
+
+    assert_equal "M", @cell.render
+  end
+
+  def test_render_is_dynamic_for_ship_placement_and_hit
+    cell_2 = Cell.new("C3")
+    cruiser = Ship.new("Cruiser", 3)
+
+    assert_equal ".", cell_2.render
+
+    cell_2.place_ship(cruiser)
+
+    assert_equal "S", cell_2.render(true)
```

Comment:
why the parameter?

Reply:
The parameter was something that was included in the interaction pattern. I'm not quite sure what the parameter is doing so we should look into this.

Reply:
Figured this out. The "true" boolean is an optional boolean that will help differentiate between  whether the space should render for the player or should not render if it will be for the computer.